plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowDefinition.java

```diff
@@ -102,6 +106,12 @@ public boolean isSandbox() {
         return sandbox;
     }
 
+    @Restricted(NoExternalUse.class) // stapler
+    public boolean shouldHideSandbox() {
+        // sandbox checkbox is shown to admins even if the global configuration says otherwise
+        return CPSConfiguration.get().isHideSandbox() && !Jenkins.get().hasPermission(Jenkins.ADMINISTER);
```

Comment:
If sandbox is already disabled, you should still show the checkbox; otherwise the user would have no way to turn it back on from the GUI.

Reply:
Not sure. If the sandbox was disabled in the job it was probaby for a good reason, so enable it back probably requires an admin intervention (to approve some signature maybe?), and that point the admin can also enable it.

Also, it would generate a weird behaviour that users won't understand: you see the checkbox, enable it, save, next time you open the job configure page there is no checkbox... weird, at least.

Reply:
Not ideal, but seems better than having a job be broken in a way that the user cannot see and cannot fix without resort to REST or CLI.

Reply:
Done in https://github.com/jenkinsci/workflow-cps-plugin/pull/907/commits/5b9eccddefb3ae6c7ac2d01dec45aeb884a1d095